docs/content/getting-started/production-best-practices.mdx

```diff
@@ -21,17 +21,53 @@ The following list outlines some guidelines and best practices for running OpenF
+
+## Concurrency Limits
+:::note
+Before modifying concurrency limits please make sure you've followed the guidance for [Database Recommendations](./#database-recommendations)
+:::
+
+OpenFGA queries such as Check and ListObjects can be quite database and CPU intensive in some cases. If you notice that a single request is consuming a lot of CPU or creating a high degree of database contention, then you may consider setting some concurrency limits to protect other requests from being negatively impacted by overly aggressive queries. 
+
+The following table enumerates the server's concurrency specific settings:
+
+| flag                                    | env                                           | config                           |
+|-----------------------------------------|-----------------------------------------------|----------------------------------|
+| --max-concurrent-reads-for-list-objects | OPENFGA_MAX_CONCURRENT_READS_FOR_LIST_OBJECTS | maxConcurrentReadsForListObjects |
+| --max-concurrent-reads-for-check        | OPENFGA_MAX_CONCURRENT_READS_FOR_CHECK        | maxConcurrentReadsForCheck       |
+| --resolve-node-limit                    | OPENFGA_RESOLVE_NODE_LIMIT                    | resolveNodeLimit                 |
+| --resolve-node-breadth-limit            | OPENFGA_RESOLVE_NODE_BREADTH_LIMIT            | resolveNodeBreadthLimit          |
+
+
+Determining the right values for these settings will be based on a variety of factors including, but not limited to, the database specific deployment topology, the FGA model(s) involved, and the relationship tuples in the system. However, here are some high-level guidelines:
+
+* If a single ListObjects query is negatively impacting other query endpoints, then consider setting a stricter value for `OPENFGA_MAX_CONCURRENT_READS_FOR_LIST_OBJECTS`.
+
+* If a single Check query is negatively impacting other query endpoints, then consider setting a stricter value for `OPENFGA_MAX_CONCURRENT_READS_FOR_CHECK`.
+
+If you still see high request latencies despite the guidance above, then you may additionally consider setting stricter limits on the query resolution behavior by limiting the resolution depth and resolution breadth. These can be controlled with the `OPENFGA_RESOLVE_NODE_LIMIT` and `OPENFGA_RESOLVE_NODE_BREADTH_LIMIT` settings, respectively. Consider these guidelines:
+
+* `OPENFGA_RESOLVE_NODE_LIMIT` limits the resolution depth of a single query, and thus it sets an upper bound on how deep a relationship hierarchy may be. A high value will allow a single query to involve more hierarchical resolution and therefore more database queries, while a low value will reduce the number of hierarchical resolutions that will be allowed and thus reduce the number of database queries.
```

Comment:
> `OPENFGA_RESOLVE_NODE_LIMIT` limits the resolution depth of a single query, and thus it sets an upper bound on how deep a relationship hierarchy may be.

Can give an example here, e.g. if a model has `define c: b` and `define b:a`, this setting puts a limit on how many of these indirections you can have

Reply:
I agree here. If we want to have another page that describes the resolution behavior of Checks and ListObjects, then I think this level of content would be a good fit for that page.

Reply:
But you're already hinting at how it works with `and thus it sets an upper bound on how deep a relationship hierarchy may be`. My suggestion is to make this explanation more clear